### PR TITLE
kernelci.org: add blog post with API timeline

### PR DESCRIPTION
**Pull request by gctucker**
Add a blog post to announce the overall timeline for transitioning to the new API & Pipeline.

### CODE REVIEW

**Comment by gctucker** [comment]
Deployed on staging: https://static.staging.kernelci.org/blog/posts/2023/api-timeline/

**Comment by gctucker** [comment]
The last part is not finished, I'll continue this in time so we can hopefully publish it on Monday.

**Comment by gctucker** [comment]
> The last part is not finished, I'll continue this in time so we can hopefully publish it on Monday.

OK that's done now, so this PR is all ready for review.  I might just do another pass tomorrow to tweak some minor things or fix typos and misspellings.

File: kernelci.org/content/en/blog/posts/2023/api-timeline/index.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+: Monday 4th December 2023
+
+Legacy System Deprecation
+: Monday 4th March 2024
+
+Legacy System Sunset
+: Monday 4th November 2024
+
+It only takes four Monday the Forth milestones to get through all this.  Here's
+what they mean:
```
Comment (JenySadadia) [inline kernelci.org/content/en/blog/posts/2023/api-timeline/index.md:82]: Just a minor thing.
```suggestion
It only takes four Mondays the Forth milestone to get through all this.  Here's
what they mean:
```

Comment (gctucker) [reply kernelci.org/content/en/blog/posts/2023/api-timeline/index.md:82]: I intentionally wrote it as "Monday the Forth" milestones.  There are definitely 4 milestones.  I think I'm just missing dashes in "Monday-the-Forth" to be completely correct from a grammatical point of view.

Comment (gctucker) [reply kernelci.org/content/en/blog/posts/2023/api-timeline/index.md]: OK I've updated this part now.

**Review by JenySadadia — APPROVED** [review]
Looks great to me.

**Comment by gctucker** [comment]
OK thanks for the reviews, this is now ready to be published.

**Pull request merged**